packages/matter-node-ble.js/src/ble/BlenoBleServer.ts

```diff
@@ -0,0 +1,352 @@
+
+            if (!error) {
+                Bleno.setServices([this.matterBleService]);
+            }
+            // TODO handle transport error
+        });
+
+        Bleno.on('advertisingStop', () => {
+            logger.debug('advertisingStop');
+        });
+
+        Bleno.on('servicesSet', (error) => {
+            logger.debug(`servicesSet: ${error ? `error ${error}` : 'success'}`);
+        });
+    }
+
+    /**
+     * Process a Write request on characteristic C1 from the Matter service.
+     * The data are checked if it might be a handshake request and stored until the subscribe request comes in.
+     * Otherwise, the data are forwarded to the BTP session handler to be decoded and processed.
+     *
+     * @param data
+     * @param offset
+     * @param withoutResponse
+     */
+    handleC1WriteRequest(data: Buffer, offset: number, withoutResponse: boolean) {
+        if (offset !== 0 || withoutResponse) {
+            throw new Error(`Offset ${offset} or withoutResponse ${withoutResponse} not supported`);
+        }
+
+        if (data[0] === 0x65 && data[1] === 0x6c && data.length === 9) { // Check if the first two bytes and length match the Matter handshake
+            this.btpHandshakeTimeout.start(); // starts timer
+
+            logger.info(`Received Matter handshake request: ${data.toString('hex')}, store until subscribe request comes in.`);
+            this.latestHandshakePayload = data;
+            // TODO Handle edge case where handshake comes with an already open BTP session (should never happen?)
+        } else {
+            if (this.btpSession !== undefined) {
+                logger.debug(`Received Matter data for BTP Session: ${data.toString('hex')}`);
+                void this.btpSession.handleIncomingBleData(new ByteArray(data));
+            } else {
+                throw new Error(`Received Matter data but no BTP session was initialized: ${data.toString('hex')}`);
+            }
+        }
+    }
+
+    /**
+     * Process a Subscribe request on characteristic C2 from the Matter service.
+     * This is expected directly after a handshake request and initializes the BTP session handler with the stored
+     * handshake payload.
+     * The BtpSessionHandler instance is wired with the bleno instance for sending data and disconnecting.
+     *
+     * @param maxValueSize
+     * @param updateValueCallback
+     */
+    async handleC2SubscribeRequest(maxValueSize: number, updateValueCallback: (data: Buffer) => void) {
+        if (this.latestHandshakePayload === undefined) {
+            throw new Error(`Subscription request received before handshake Request`);
+        }
+        if (this.btpSession !== undefined) {
+            throw new Error(`Subscription request received but BTP session already initialized. Can not handle two sessions!`);
+        }
+        this.btpHandshakeTimeout.stop();
+
+        this.btpSession = await BtpSessionHandler.createFromHandshakeRequest(
```

Comment:
An object argument could make the arguments clearer here

Reply:
maybe a bit, but in fact IDE should support both